### PR TITLE
Handle client-side errors on debug pod creation

### PR DESCRIPTION
**Pull request by fabianofranz**
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1330050

ptal @smarterclayton

### CODE REVIEW

**Comment by fabianofranz** [comment]
[test]

File: pkg/cmd/cli/cmd/debug.go [hunk 1]
```diff
@@ -406,10 +406,13 @@ func (o *DebugOptions) Debug() error {
 		func() {
 			fmt.Fprintf(o.Attach.Err, "\nRemoving debug pod ...\n")
 			if err := o.Attach.Client.Pods(pod.Namespace).Delete(pod.Name, kapi.NewDeleteOptions(0)); err != nil {
-				fmt.Fprintf(o.Attach.Err, "error: unable to delete the debug pod %q: %v", pod.Name, err)
+				if !kapierrors.IsNotFound(err) {
+					fmt.Fprintf(o.Attach.Err, "error: unable to delete the debug pod %q: %v\n", pod.Name, err)
```
Comment (smarterclayton) [inline pkg/cmd/cli/cmd/debug.go]: Do we even need to print a message?

File: pkg/cmd/cli/cmd/debug.go [hunk 2]
```diff
@@ -418,6 +421,13 @@ func (o *DebugOptions) Debug() error {
 		}
 		fmt.Fprintf(o.Attach.Err, "Waiting for pod to start ...\n")
 		switch _, err := Until(o.Timeout, w, PodContainerRunning(o.Attach.ContainerName)); {
+		// api didn't error right away but the pod wasn't even created
+		case kapierrors.IsNotFound(err):
+			msg := fmt.Sprintf("error: unable to create the debug pod %q", pod.Name)
```
Comment (smarterclayton) [inline pkg/cmd/cli/cmd/debug.go]: should not be error prefixed

Comment (fabianofranz) [inline pkg/cmd/cli/cmd/debug.go]: of course, fixed.

File: pkg/cmd/cli/cmd/debug.go (hunk 1 above)
Comment (fabianofranz) [inline pkg/cmd/cli/cmd/debug.go]: I think so, for errors other than pod not found.

Comment (fabianofranz) [inline pkg/cmd/cli/cmd/debug.go]: You could successfully run a debug session, and when exiting the debug pod fails to be removed for any other reason - you should know the command left some garbage behind.

**Comment by fabianofranz** [comment]
@smarterclayton would you like to get this in?

**Comment by smarterclayton** [comment]
LGTM [merge]

**Pull request merged**